librosa/core/pitch.py

```diff
@@ -333,3 +338,446 @@ def piptrack(y=None, sr=22050, S=None, n_fft=2048, hop_length=None,
+        shape parameters for the beta distribution prior over thresholds.
+
+    boltzmann_parameter: number > 0 [scalar]
+        shape parameter for the Boltzmann distribution prior over troughs.
+        Larger values will assign more mass to smaller periods.
+
+    resolution : float in `(0, 1)`
+        Resolution of the pitch bins.
+        0.01 corresponds to cents.
+
+    max_transition_rate : float > 0
+        maximum pitch transition rate in octaves per second.
+
+    switch_prob : float in ``(0, 1)``
+        probability of switching from voiced to unvoiced or vice versa.
+
+    no_trough_prob : float in ``(0, 1)``
+        maximum probability to add to global minimum if no trough is below threshold.
+
+    center : boolean
+        If ``True``, the signal ``y`` is padded so that frame
+        ``D[:, t]`` is centered at ``y[t * hop_length]``.
+        If ``False``, then ``D[:, t]`` begins at ``y[t * hop_length]``.
+        Defaults to ``True``,  which simplifies the alignment of ``D`` onto a
+        time grid by means of ``librosa.core.frames_to_samples``.
+
+    pad_mode : string or function
+        If ``center=True``, this argument is passed to ``np.pad`` for padding
+        the edges of the signal ``y``. By default (``pad_mode="reflect"``),
+        ``y`` is padded on both sides with its own reflection, mirrored around
+        its first and last sample respectively.
+        If ``center=False``,  this argument is ignored.
+        .. see also:: `np.pad`
+
+    Returns
+    -------
+    f0: np.ndarray [shape=(n_frames,)]
+        time series of fundamental frequencies in Hertz.
+
+    voiced_flag: np.ndarray [shape=(n_frames,)]
+        time series containing boolean flags indicating whether a frame is voiced or not.
+
+    voiced_prob: np.ndarray [shape=(n_frames,)]
+        time series containing the probability that a frame is voiced.
+
+    See Also
+    --------
+    librosa.yin
+        Fundamental frequency (F0) estimation using the YIN algorithm.
+
+    Examples
+    --------
+    Computing a fundamental frequency (F0) curve from an audio input
+
+    >>> y, sr = librosa.load(librosa.ex('trumpet'))
+    >>> f0, voiced_flag, voiced_probs = librosa.pyin(y, fmin=librosa.note_to_hz('C2'), fmax=librosa.note_to_hz('C7'))
+    >>> times = librosa.times_like(f0)
+
+
+    Overlay F0 over a spectrogram
+
+    >>> import matplotlib.pyplot as plt
+    >>> D = librosa.amplitude_to_db(np.abs(librosa.stft(y)), ref=np.max)
+    >>> fig, ax = plt.subplots()
+    >>> img = librosa.display.specshow(D, x_axis='time', y_axis='log', ax=ax)
+    >>> ax.set(title='pYIN fundamental frequency estimation')
+    >>> fig.colorbar(img, ax=ax, format="%+2.f dB")
+    >>> ax.plot(times, np.where(voiced_flag, f0, np.nan), label='f0', color='w')  # mask unvoiced frames with np.nan
+    >>> ax.legend(loc='upper right')
+    '''
+    if win_length > frame_length:
+        raise ParameterError('win_length cannot exceed frame_length')
```

Comment:
can we default `win_length=None` and infer `frame_length//2` if it's not explicitly set?